src/main.c

```diff
@@ -267,25 +267,67 @@ static gboolean checksum_start(int argc, char **argv)
 	return TRUE;
 }
 
+/* Takes a shell variable and its desired argument as input and appends it to
+ * the provided text with taking care of correct shell quoting */
+static void info_formatter_shell_append(GString* text, const gchar* varname, const gchar* argument) {
+	gchar* quoted = g_shell_quote (argument ?: "");
+	g_string_append_printf(text, "%s=%s\n", varname, quoted);
+	g_clear_pointer(&quoted, g_free);
+}
+/* Same as above, expect that it has a cnt argument to add per-slot-number
+ * strings */
+static void info_formatter_shell_append_n(GString* text, const gchar* varname, gint cnt, const gchar* argument) {
+	gchar* quoted = g_shell_quote (argument ?: "");
+	g_string_append_printf(text, "%s_%d=%s\n", varname, cnt, quoted);
+	g_clear_pointer(&quoted, g_free);
+}
+
 static gchar *info_formatter_shell(RaucManifest *manifest)
 {
 	GString *text = g_string_new(NULL);
+	GPtrArray *hooks = NULL;
 	gint cnt;
 
-	g_string_append_printf(text, "RAUC_MF_COMPATIBLE=%s\n", manifest->update_compatible);
-	g_string_append_printf(text, "RAUC_MF_VERSION=%s\n", manifest->update_version ?: "");
-	g_string_append_printf(text, "RAUC_MF_DESCRIPTION=%s\n", manifest->update_description ?: "");
-	g_string_append_printf(text, "RAUC_MF_BUILD=%s\n", manifest->update_build ?: "");
+	info_formatter_shell_append(text, "RAUC_MF_COMPATIBLE", manifest->update_compatible);
+	info_formatter_shell_append(text, "RAUC_MF_VERSION", manifest->update_version ?: "");
+	info_formatter_shell_append(text, "RAUC_MF_DESCRIPTION", manifest->update_description);
+	info_formatter_shell_append(text, "RAUC_MF_BUILD", manifest->update_build);
 	g_string_append_printf(text, "RAUC_MF_IMAGES=%d\n", g_list_length(manifest->images));
 	g_string_append_printf(text, "RAUC_MF_FILES=%d\n", g_list_length(manifest->files));
 
+	hooks = g_ptr_array_new();
+	if (manifest->hooks.install_check == TRUE) {
+		g_ptr_array_add(hooks, g_strdup("install-check"));
+	}
+	g_ptr_array_add(hooks, NULL);
+
+	info_formatter_shell_append(text, "RAUC_MF_HOOKS", g_strjoinv(" ", (gchar**) hooks->pdata));
```

Comment:
This leaks the result of g_strjoinv. See also below. Make a helper function to append a space separated string array?